rust/candid/src/de.rs

```diff
@@ -258,10 +258,13 @@ impl<'de> Deserializer<'de> {
     fn is_zero_sized_type(&self, t: &Type) -> bool {
         match t.as_ref() {
             TypeInner::Null | TypeInner::Reserved => true,
-            TypeInner::Record(fs) => fs.iter().all(|f| {
-                let t = self.table.trace_type(&f.ty).unwrap();
-                self.is_zero_sized_type(&t)
-            }),
+            TypeInner::Record(fs) => {
+                fs.is_empty()
+                    || fs.iter().all(|f| {
+                        let t = self.table.trace_type(&f.ty).unwrap();
+                        self.is_zero_sized_type(&t)
+                    })
```

Comment:
No need to special case the empty record, `.all` does the right thing there too.